modules/directives/src/main/scala/scala/build/preprocessing/directives/DirectiveHandler.scala

```diff
@@ -15,10 +27,136 @@ trait DirectiveHandler[T] {
+    * [[unexpectedValueHint]] inside those handlers.
+    *
+    * @param scopedDirective
+    *   the directive whose values are to be processed according to the key
+    * @return
+    *   Either a [[BuildException]] or the grouped using directive values inside the passed
+    *   [[scopedDirective]]
+    */
+  final protected def checkIfValuesAreExpected(
+    scopedDirective: ScopedDirective
+  ): Either[BuildException, GroupedScopedValuesContainer] =
+    receiveTheRightNumberOfValues(scopedDirective) flatMap (
+      siftSupportedValueTypesWithNoResidue(_)
+    )
+
+  /** the default implementation only includes `UsingDirectiveValueKind.STRING` Override for
+    * changing it.
+    * @param key
+    *   the using directive key for which the supported types are calculated.
+    * @return
+    *   a set of supported using directive value kinds of string, numeric, or boolean by this
+    *   handler
+    */
+  protected def getSupportedTypes(key: String): Set[UsingDirectiveValueKind] =
+    Set(UsingDirectiveValueKind.STRING)
+
+  /** by default, it is `(lower = 1, upper = Int.MaxValue)`. Please, override for other cases.
+    * @param key
+    *   the using directive key for which the value number bounds are calculated.
+    */
+  protected def getValueNumberBounds(key: String): UsingDirectiveValueNumberBounds =
+    UsingDirectiveValueNumberBounds()
+
+  /** The default implementation just asks the user to put quotation marks around the using
+    * directive values with string type.
+    *
+    * Therefore, it should get overrided if other information is to be shown instead in any
+    * particular handler.
+    *
+    * This method gets internally called in the implementation of
+    * [[siftSupportedValueTypesWithNoResidue]] for producing the error value.
+    * @param key
+    *   the using directive key for which the unexpected value(s) are passed.
+    * @return
+    *   the hint to be shown to the user if the value(s) passed to the corresponding using directive
+    *   key are unexpected.
+    */
+  protected def unexpectedValueHint(key: String): String =
+    s"Did you forget to put the quotation marks around the string values passed to the using directive key $key?"
+
+  /** checks if the passed in `scopedDirective` has the expected number of values according to the
+    * implementation of [[getValueNumberBounds]] for the handler on which this is called
+    *
+    * @return
+    *   Either a [[UsingDirectiveExpectationError]] or the same passed in scopedDirective
+    */
+  final protected def receiveTheRightNumberOfValues(
+    scopedDirective: ScopedDirective
+  ): Either[UsingDirectiveExpectationError, ScopedDirective] = {
+    val values: Seq[Value[_]] = scopedDirective.directive.values.collect {
+      case n: NumericValue => n
+      case b: BooleanValue =>
+        b
+      case s: StringValue => s
+    } // TODO or perhaps `.filterNot(_.isInstanceOf[EmptyValue])` ??
```

Comment:
I think `filterNot` is better apprach, especially combined with a function `isEmptyValue(v: Value) = v isInstanceOf[EmptyValue]`